documentation/src/main/asciidoc/reference/search-dsl-highlighting.asciidoc

```diff
@@ -0,0 +1,473 @@
+====
+
+[[search-dsl-highlighting-highlighter-options]][[search-dsl-highlighting-highlighter-options-tags]]
+== Tags
+
+By default, the highlighted text is wrapped with a pair of `<em>`/`</em>` tags. A custom pair of tags can be provided to change this behaviour.
+Usually, tags are a pair of HTML tags, but they can be a pair of any character sequences.
+
+.Setting custom tags
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=basic-tags]
+----
+<1> Passing a pair of open/close tags that will be used to highlight the text.
+====
+
+The fast vector highlighter, which can handle multiple tags, has a few additional methods that accept a collection of tags.
+
+.Setting multiple custom tags
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=fvh-tags]
+----
+<1> Passing a collection of open tags and a single closing tag that will be used to highlight the text.
+It can be helpful when configuring a tag schema that differs only in an attribute of an opening tag.
+<2> Passing a pair of collections containing open/close tags that will be used to highlight the text.
+====
+
+[NOTE]
+====
+Calling different tags configuration methods or the same one multiple times
+within the same highlighter definition will *not* combine them.
+Tags set by the last call will be applied.
+====
+
+Additionally, a fast vector highlighter has the option to enable a tag schema and set it to `HighlighterTagSchema.STYLED`
+to use a predefined set of tags.
+
+.Setting a styled tags schema
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=fvh-schema]
+----
+<1> Passing a styled tags schema that will be used to highlight the text.
+====
+
+Using a styled tags schema is just a shortcut to defining tags as:
+
+.Setting tags as if the styled tags schema is used
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=fvh-schema-alternative]
+----
+<1> Passing the same collection of tags used when a styled schema is applied.
+====
+
+[NOTE]
+====
+If a styled tags schema is applied to a highlighter, then any other tag configurations are ignored, and the schema tags are used.
+====
+
+[[search-dsl-highlighting-highlighter-options-encoder]]
+== Encoder
+Encoding can be applied to the highlighted snippets when highlighting the fields that store HTML. Applying an
+HTML encoder to a highlighter will encode the text for inclusion into an HTML document:
+it will replace HTML meta-characters such as `<` with their entity equivalent such as `&amplt;`;
```

Comment:
For some reason this triggers a build error when generating PDF documentation (`-Pdocumentation-pdf`).

I think there's a missing `;`?

```suggestion
it will replace HTML meta-characters such as `<` with their entity equivalent such as `&amp;lt;`;
```

Alternatively, maybe you should use the `+` notation:


```suggestion
it will replace HTML meta-characters such as `<` with their entity equivalent such as `+&lt;+`;
```